apps/jobbrowser/src/jobbrowser/api2.py

```diff
@@ -216,16 +188,58 @@ def query_store_api(request, path=None):
 
 @api_error_handler
 def query_store_download_bundle(request, id=None):
-  response = {}
+  client = _create_query_store_client(request, content_type='application/octet-stream')
 
-  client = HttpClient(QUERY_STORE.SERVER_URL.get())
+  try:
+    zip_resp = client.execute("GET", 'api/data-bundle/' + id, stream=True);
+    response = StreamingHttpResponse(_zip_generate(zip_resp), content_type=zip_resp.headers.get('Content-Type'))
+    response['Content-Disposition'] = zip_resp.headers.get('Content-Disposition')
+    return response
+  except RestException as e:
+    return _handle_query_store_exception(e)
+
+
+def _query_store_proxy(request, path=None):
```

Comment:
BTW: would it make sense to move all the query store specific utils (i.e. everything but the views, the _utils()..) into a apis/query_store.py so that we keep it cleaner and not mixed with the common API interface? 

It will be simpler to add tests this way too.

Reply:
Was thinking about moving all of these non-view functions into a separate file, but wasn't sure where to put them. Thanks for the comment, will make the changes as suggested.

Reply:
I have updated the patch with two new file
- query_store - That houses all the query store related code
- stream - A generic streaming api